### ISSUE TITLE
Cannot use gitignored .env.local in a runtime sharedGlobal

### ISSUE THREAD

**Issue by martin-mindflow** [nrwl/nx#19172]
### Current Behavior

I have an nx monorepo, with several libs and apps. I want the `.env.local` file to be added to the cache hashes, so that all caches are invalidated when this file changes. The file is gitignored, so i can't just put its path in a sharedGlobal, seems like files inside of .gitignore are hard ignored by nx.

So, i use the following conf as a workaround: 

```
    "sharedGlobals": [
      {
        "runtime": "cat .env.local"
      }
    ]
```

which does not work. ie:

```
nx lint debug # this is a cache miss, as expected
nx lint debug # cache hit, as expected
echo "whatever" >> .env.local
nx lint debug # this should be a miss but it's a hit :(
```

which lead me to believe my approach was wrong, but inexplicably, when i use another file in the conf:

```
    "sharedGlobals": [
      {
        "runtime": "cat file.txt"
      }
    ]
```

```
nx lint debug # this is a cache miss, as expected
nx lint debug # cache hit, as expected
echo "whatever" >> file.txt
nx lint debug # cache miss!
```

### Expected Behavior

I'm expecting the cache to behave the same way for both files

### GitHub Repo

https://github.com/martin-mindflow/nx-cache-reproduction

### Steps to Reproduce

1. clone the example repo on the main branch
2. run the following commands sequence:
```
nx lint debug # this is a cache miss, as expected
nx lint debug # cache hit, as expected
echo "whatever" >> .env.local
nx lint debug # this should be a miss but it's a hit :(
```
3. switch to the `ok` branch
4. run the following commands sequence:
```
nx lint debug # this is a cache miss, as expected
nx lint debug # cache hit, as expected
echo "whatever" >> file.txt
nx lint debug # cache miss!
```

### Nx Report

```shell
Node   : 18.16.1
   OS     : darwin-arm64
   npm    : 9.5.1

   nx                 : 16.8.1
   @nx/js             : 16.8.1
   @nx/linter         : 16.8.1
   @nx/workspace      : 16.8.1
   @nx/devkit         : 16.8.1
   @nx/eslint-plugin  : 16.8.1
   @nx/react          : 16.8.1
   @nrwl/tao          : 16.8.1
   @nx/vite           : 16.8.1
   @nx/web            : 16.8.1
   typescript         : 5.1.6
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by Chanki-Min** [comment]
releated: https://github.com/nrwl/nx/issues/19392

**Comment by Cammisuli** [comment]
There's a way to make this work for your scenario, but while I was testing it, I found an issue 😅 

Anyway, once the linked PR is merged, you'll be able to properly use the `.nxignore` file. Adding `!.env.local` to that file will whitelist it, and it will be included in hashes.

Usually folks ignored whole folders, while `nxignore`ing subfolders, so the logic worked well there. With this scenario, we ignored and whitelisted the same file, so things got weird.
